File: pgm/ABC_Java/ABC_313_A.java
import java.io.*;
// import java.math.*;
import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        reader.readLine();
        final int[] P = Arrays.stream (reader.readLine().split (" "))
                            .mapToInt (Integer::parseInt)
                            .toArray();
        final int mx = Arrays.stream (P).max().orElse (0);
        final long cnt = Arrays.stream (P).filter (p -> p == mx).count();
        System.out.println (
            (P[0] != mx || (P[0] == mx && cnt > 1)) ? mx - P[0] + 1 : 0);
        return;
    }
}

File: pgm/ABC_Kotlin/ABC_313_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.time.*
// import java.util.StringTokenizer

// import kotlin.math.*

const val space: String = " "
const val one: Int = 1
const val zero: Int = 0

fun main() {
    readln().toInt()
    val P: IntArray = readln().split(space).map { it.toInt() }.toIntArray()
    val max: Int = P.maxOrNull() ?: 0
    val cnt: Int = P.count {it == max}
    if (P[zero] != max || (P[zero] == max && cnt > one)) {
        println(max - P[zero] + one)
    } else {
        println(zero)
    }
    kotlin.system.exitProcess(0)
}